charts/fybrik/files/taxonomy/fybrik_module.json

```diff
@@ -0,0 +1,70 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "$id": "fybrik_module.json",
+    "title": "FybrikModule taxonomy",
+    "type": "object",
+    "description": "Values contributed to by Fybrik modules.",
+    "properties": {
+        "spec": {
+            "type": "object",
+            "properties": {
+                "pluginType": {
+                    "$ref": "taxonomy.json#/definitions/PluginType"
+                },
+                "capabilities": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "properties": {
+		                    "capability": {
+                                "$ref": "taxonomy.json#/definitions/Capability"
+                            },
+                            "api": {
+                                "type": "object",
+                                "properties": {
+                                    "interfaceDetails": {
+                                        "$ref": "taxonomy.json#/definitions/Interface"
+                                    }
+                                },
+                                "required": ["interfaceDetails"]
+                            },
+                            "supportedInterfaces": {
+                                "type": "array",
+                                "items": {
+                                    "type": "object",
+                                    "oneOf": [
+                                        {
+                                            "source": {
+                                                "$ref": "taxonomy.json#/definitions/Interface"
+                                            }
+                                        },
+                                        {
+                                            "sink": {
+                                                "$ref": "taxonomy.json#/definitions/Interface"
+                                            }
+                                        }
+                                    ]
```

Comment:
I don't think that it's oneOf. See for example https://github.com/fybrik/fybrik/blob/6138f68ad33ffc6f9179f58f29bf33b804939437/modules/implicit-copy-stream-module.yaml#L17-L22. 

@shlomitk1 why is supportedInterfaces defined as an object array in the CRD? What's the rational behind not making it a concrete type?

Reply:
@roee88 Not sure I understand. What is the alternative - to create an intermediate type that holds the same list of <source,sink> or changing the structure entirely?

Reply:
> to create an intermediate type that holds the same list of <source,sink>

For example. Would that be ok?

Reply:
Yes.